scanner/src/test/kotlin/scanners/fossid/FossIdTest.kt

```diff
@@ -0,0 +1,829 @@
+            val pkg2 = createPackage(id2, createVcsInfo("anotherProject"), authors = pkgAuthors)
+
+            val config = createConfig(packageNamespaceFilter = id1.namespace, packageAuthorsFilter = ignoredAuthor)
+            val fossId = createFossId(config)
+
+            val scannerRun = fossId.scan(listOf(pkg1, pkg2))
+            scannerRun.results.collectIssues().keys should beEmpty()
+        }
+
+        "create a delta scan for an existing scan" {
+            val projectCode = projectCode(PROJECT)
+            val originCode = "originalScanCode"
+            val scanCode = scanCode(PROJECT, FossId.DeltaTag.DELTA)
+            val config = createConfig()
+            val vcsInfo = createVcsInfo()
+            val scan = createScan(vcsInfo.url, vcsInfo.revision, originCode)
+
+            val service = config.createService()
+                .expectProjectRequest(projectCode)
+                .expectListScans(projectCode, listOf(scan))
+                .expectCheckScanStatus(originCode, ScanState.FINISHED)
+                .expectCheckScanStatus(scanCode, ScanState.NOT_STARTED, ScanState.FINISHED)
+                .expectCreateScan(projectCode, scanCode, vcsInfo)
+                .expectDownload(scanCode)
+                .mockFiles(scanCode)
+            coEvery { service.runScan(any()) } returns EntityResponseBody(status = 200)
+
+            val fossId = createFossId(config)
+
+            fossId.scan(listOf(createPackage(createIdentifier(index = 1), vcsInfo)))
+
+            coVerify {
+                service.createScan(USER, API_KEY, projectCode, scanCode, vcsInfo.url, vcsInfo.revision)
+                service.downloadFromGit(USER, API_KEY, scanCode)
+                service.checkDownloadStatus(USER, API_KEY, scanCode)
+                service.runScan(USER, API_KEY, scanCode, *FossId.deltaScanRunParameters(originCode))
+            }
+        }
+
+        "create a delta scan if no scan exists yet" {
+            val projectCode = projectCode(PROJECT)
+            val scanCode = scanCode(PROJECT, FossId.DeltaTag.ORIGIN)
+            val config = createConfig()
+            val vcsInfo = createVcsInfo()
+
+            val service = config.createService()
+                .expectProjectRequest(projectCode)
+                .expectListScans(projectCode, emptyList())
+                .expectCheckScanStatus(scanCode, ScanState.FINISHED)
+                .expectCreateScan(projectCode, scanCode, vcsInfo)
+                .expectDownload(scanCode)
+                .mockFiles(scanCode, identifiedRange = 1..2, markedRange = 1..2)
+
+            val fossId = createFossId(config)
+
+            fossId.scan(listOf(createPackage(createIdentifier(index = 1), vcsInfo)))
+
+            coVerify {
+                service.createScan(USER, API_KEY, projectCode, scanCode, vcsInfo.url, vcsInfo.revision)
+                service.downloadFromGit(USER, API_KEY, scanCode)
+                service.checkDownloadStatus(USER, API_KEY, scanCode)
```

Comment:
Can you also test that `runScan` was NOT called with delta scan run parameters ?